messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/ServerAdd.java

```diff
@@ -456,15 +456,23 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
             ServiceController activeMQServerServiceController = serviceBuilder.setInstance(serverService)
                     .install();
             //Add the queue services for the core queues  created throught the internal broker configuration (those queues are not added as service via the QueueAdd OSH)
-            for (CoreQueueConfiguration queueConfiguration : configuration.getQueueConfigurations()) {
-                final ServiceName queueServiceName = activeMQServiceName.append(queueConfiguration.getName());
-                final ServiceBuilder sb = context.getServiceTarget().addService(queueServiceName);
-                sb.requires(ActiveMQActivationService.getServiceName(activeMQServiceName));
-                Supplier<ActiveMQServer> serverSupplier = sb.requires(activeMQServiceName);
-                final QueueService queueService = new QueueService(serverSupplier, queueConfiguration, false, false);
-                sb.setInitialMode(Mode.PASSIVE);
-                sb.setInstance(queueService);
-                sb.install();
+            if (model.hasDefined(CommonAttributes.QUEUE)) {
+                Set<String> coreQueues = new HashSet<>();
+                for (Property prop : model.get(CommonAttributes.QUEUE).asPropertyList()) {
+                    coreQueues.add(prop.getName());
+                }
```

Comment:
You don't need to copy out the names; you can just do 

ModelNode coreQueues = model.get(CommonAttributes.QUEUE);

And then 

if (coreQueues.has(queueConfiguration.getName())) {